cstrml/swork/src/lib.rs

```diff
@@ -577,103 +626,79 @@ impl<T: Trait> Module<T> {
+        let total_free = Self::free().saturating_sub(old_free as u128).saturating_add(reported_srd_size as u128);
 
         Used::put(total_used);
         Free::put(total_free);
     }
 
     /// Update sOrder information based on changed files, return the real changed files
-    fn update_sorder(reporter: &T::AccountId, changed_files: &Vec<(MerkleRoot, u64)>, is_added: bool) -> Vec<(MerkleRoot, u64)> {
+    fn update_files(
+        reporter: &T::AccountId,
+        changed_files: &Vec<(MerkleRoot, u64)>,
+        anchor: &SworkerPubKey,
+        is_added: bool) -> Vec<(MerkleRoot, u64)> {
         let mut real_files = vec![];
-        let current_block_numeric = Self::get_current_block_number();
-
-        if let Some(mi) = T::MarketInterface::merchants(reporter) {
-            let file_map = mi.file_map;
-
-            // 1. Loop changed files
-            real_files = changed_files.iter().filter_map(|(f_id, size)| {
-                // 2. If mapping to storage orders
-                if let Some(sorder_ids) = file_map.get(f_id) {
-                    // a. Loop storage orders(same file)
-                    for sorder_id in sorder_ids {
-                        if let Some(mut so_status) = T::MarketInterface::maybe_get_sorder_status(sorder_id) {
-                            if so_status.status != OrderStatus::Pending && current_block_numeric > so_status.expired_on {
-                                continue;
-                            }
-                            // b. Change sOrder status
-                            if !is_added {
-                                so_status.status = OrderStatus::Failed;
-                            } else if is_added && so_status.status == OrderStatus::Pending {
-                                // go panic if `current_block_numeric` > `created_on`
-                                so_status.expired_on += current_block_numeric - so_status.completed_on;
-                                so_status.completed_on = current_block_numeric;
-                                so_status.claimed_at = current_block_numeric;
-                                so_status.status = OrderStatus::Success;
-                            } else {
-                                so_status.status = OrderStatus::Success;
-                            }
-                            // c. Set sOrder status
-                            T::MarketInterface::maybe_set_sorder_status(sorder_id, &so_status, &current_block_numeric);
-                        }
+        let curr_bn = Self::get_current_block_number();
+
+        // 1. Loop changed files
+        if is_added {
+            real_files = changed_files.iter().filter_map(|(cid, size)| {
+                let mut is_counted = true;
+                if let Some(identity) = Self::identities(reporter) {
+                    if let Some(owner) = identity.group {
+                        is_counted = !T::MarketInterface::check_duplicate_in_group(cid, &Self::groups(owner));
```

Comment:
merge `check_duplicate_in_group` and `upsert_payouts`

Reply:
good suggestion. Done